EIPS/eip-3416.md

```diff
@@ -0,0 +1,114 @@
+* **Inefficiencies of first price auctions**: The current approach, where transaction senders publish a transaction with a bid a maximum fee, miners choose the highest-paying transactions, and everyone pays what they bid. This is well-known in mechanism design literature to be highly inefficient, and so complex fee estimation algorithms are required. But even these algorithms often end up not working very well, leading to frequent fee overpayment. We need a more stable fee metric that is computed inside the protocol.
+
+The proposal in this EIP is to start with a base fee amount which is adjusted up and down by the protocol based on how congested the network is. When the network exceeds the target per-block gas usage, the base fee increases slightly and when capacity is below the target, it decreases slightly. Because these base fee changes are constrained, the maximum difference in base fee from block to block is predictable. This then allows wallets to auto-set the gas fees for users in a highly reliable fashion. It is expected that most users will not have to manually adjust gas fees, even in periods of high network activity. For most users the base fee will be estimated by their wallet and a small gas premium related to the urgency and the priority they want to instill into the transaction. 
+
+An important aspect of this fee system is that miners get to keep the median gas premium. There are several arguments in favor of miners continuing to earn base fees, mainly related to fairness of smaller miners already invested in mining hardware.
+
+The transition to this gas price system will occur in two phases, in the first phase both legacy and EIP3416 transactions will be accepted by the protocol. Over the course of this first phase the amount of gas available for processing legacy transactions will decrease while the amount of gas available for processing EIP3416 transactions will increase, moving gas from the legacy pool into the EIP3416 pool until the legacy pool is depleted and the EIP3416 pool contains the entire gas maximum. After all of the gas has transitioned to the EIP3416 pool, the second, finalized, phase is entered and legacy transactions will no longer be accepted on the network.
+
+## Specification
+
+### Definitions
+
+This is a classic fork without a long migration time.
+
+* `FORK_BLOCK_NUMBER`: TBD.  Block number at or after which EIP-3416 transactions are valid.
+* `BLOCK_GAS_TARGET`: 10th item in the block header: The gas limit field, controlled by miner voting.  Previously referred to colloquially as `gas limit`, now referred to as `gas target`.  Controlled by miners in the same way as before where each miner can increase or decrease it by a very small amount relative to the parent block.
```

Comment:
This constant isn't used elsewhere in the specification.  Is there no block elasticity?  If not, remove this constant (and you'll also need to adjust some other parts of this EIP which imply elasticity).

Reply:
I want to include *block elasticity*, will improve the specs. Thanks a lot.